### ISSUE TITLE
Generation of Wrong Multi-Extension FITS with missing END

### ISSUE THREAD

**Issue by ppmim** [astropy/astropy#5633]
Hi, I'm having trouble modifying [this](http://www.iaa.es/~jmiguel/PANIC/test.fits) Multi-Extension FITS file with 4 extensions (warning: 64MB). Even this minimal script:

```
from astropy.io import fits
hdulist = fits.open('test.fits', 'update')
hdulist[0].header.set("MYKEY", "myvalue")
hdulist.close()
```

produces a new Multi-Extension FITS header with a missing end card in the first extension. In addition, the file size is doubled, what should not happen. 

However, if I run the same script using pyfits (v3.3) instead of astropy.io.fits, it works fine and produces a valid Mult-Extension FITS and with the right size (64MB).

Can anybody reproduce this error and let me know what causes it?

I am using astropy 1.2.1, numpy 1.11.1 and python 2.7.6 on a Ubuntu 14.04.2 LTS system.

**Comment by eteq** [comment]
I can't reproduce this on mac OS 10.11.6 with the same version of astropy/numpy and python 2.7.12 (nor with py 3.5.2).  What I *do* see is that the file appears to change substantially because I *think* it's crossing the 2880-char limit.  But it still ends up with 4 END cards.

@ppmim - any chance you can try this on py 2.7.13 (or 2.7.12)?  That seems to be the only thing substantially different other than the system architectures... Also, are you on a 32-bit or 64-bit machine? (mine is 64, although I can't really see why that would matter...)

**Comment by ppmim** [comment]
@eteq With python 2.7.6, I have updated astropy/numpy to 1.3/1.11.3 and now the problem has disappeared. In addition, I have checked the problem does not happen neither with astropy 0.3 nor 1.0.6. 

So, it seems the problem is in astropy 1.1.x or 1.2.x.

I have installed (with virtualenv) python 2.7.12 and astropy 1.3, and it works also right. 

All test were done in 64-bit linux machines.

**Comment by pllim** [comment]
Unless someone can identify the specific PR that fixed this for 1.3 (nothing obvious jumped out when I searched), the only option is to add this to "known issues" for 1.1.x and 1.2.x series. Maybe something like, "If you encounter this problem, you need to upgrade to 1.3 or later."

**Comment by saimn** [comment]
Closing as the issue concerns only old versions (1.1 or 1.2) which are no more supported.
